sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -98,49 +98,40 @@ namespace Azure { namespace Data { namespace Tables {
         Core::Context const& context = {});
 
     /**
-     * @brief Update entity in a table.
-     *
-     * @param tableEntity The TableEntity to set.
-     * @param context for canceling long running operations.
-     * @return Update entity result.
-     */
-    Response<Models::UpdateEntityResult> UpdateEntity(
-        Models::TableEntity const& tableEntity,
-        Core::Context const& context = {});
-
-    /**
-     * @brief Merge entity in a table.
+     * @brief Deletes the specified entity in a table.
      *
-     * @param tableEntity The TableEntity to merge.
+     * @param tableEntity The TableEntity to delete.
      * @param context for canceling long running operations.
-     * @return Merge entity result.
+     * @return Delete entity result.
      */
-    Response<Models::MergeEntityResult> MergeEntity(
+    Response<Models::DeleteEntityResult> DeleteEntity(
         Models::TableEntity const& tableEntity,
         Core::Context const& context = {});
 
     /**
-     * @brief Deletes the specified entity in a table.
+     * @brief Update entity in a table.
      *
-     * @param tableEntity The TableEntity to delete.
+     * @param tableEntity The TableEntity to set.
+     * @param updateMode Update mode(update/merge), default merge.
      * @param context for canceling long running operations.
-     * @return Delete entity result.
+     * @return Update entity result.
      */
-    Response<Models::DeleteEntityResult> DeleteEntity(
+    Response<Models::UpdateEntityResult> UpdateEntity(
         Models::TableEntity const& tableEntity,
+        Models::UpdateMode updateMode = Models::UpdateMode::Merge,
         Core::Context const& context = {});
 
     /**
      * @brief Upsert specified entity in a table.
      *
      * @param tableEntity The TableEntity to upsert.
-     * @param upsertKind Upsert kind(update/merge), default update.
+     * @param updateMode Upsert mode(update/merge), default merge.
```

Comment:
nit: Update instead of Upsert in the comment?
```suggestion
     * @param updateMode Update mode(update/merge), default merge.
```